src/TagHelperPack/Internal/ModelHtmlHelper.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+#if !NET471
+using Microsoft.AspNetCore.Mvc.ViewFeatures.Buffers;
+#else
+using Microsoft.AspNetCore.Mvc.ViewFeatures.Internal;
+#endif
+
+namespace TagHelperPack.Internal
+{
+    /// <summary>
+    /// An <see cref="HtmlHelper"/> that implements <see cref="IModelHtmlHelper"/>.
+    /// </summary>
+    public class ModelHtmlHelper : HtmlHelper, IModelHtmlHelper
```

Comment:
Does this actually need to be public? Could it just be actually internal (rather than just in name)?

Reply:
The interface does; the class does not. But it seems pretty harmless, tucked away in a namespace that won't be imported unless folks have a specific customization in mind. I've seen this pattern a lot in AspNetCore.

But glad to change if you prefer.

Reply:
Why does the interface need to be public? Where is it needed by consumers?

We've moved away from that pattern a bit in ASP.NET Core. I'd prefer to just keep these kinds of type actually internal.

Reply:
If someone were to replace `HtmlHelper` for whatever reason, they would want their replacement to be able to implement `IModelHtmlHelper`, too. Open-Closed Principle!

Reply:
OK fair enough. Let's take it out of the `Internal` namespace then (it's either in or out).